pkg/kubelet/rkt/rkt.go

```diff
@@ -863,11 +898,34 @@ func (r *Runtime) generateRunCommand(pod *api.Pod, uuid string) (string, error)
 			return "", err
 		}
 	}
+
 	runPrepared = append(runPrepared, fmt.Sprintf("--hostname=%s", hostname))
 	runPrepared = append(runPrepared, uuid)
 	return strings.Join(runPrepared, " "), nil
 }
 
+func (r *Runtime) cleanupPodNetwork(pod *kubecontainer.Pod, teardown bool) error {
+	// Parse first container ID for the pod UUID
+	podID, err := parseContainerID(pod.Containers[0].ID)
+	if err != nil {
+		return fmt.Errorf("rkt: Failed to get rkt uuid of the pod %q: %v", pod.Name, err)
+	}
+
+	if teardown {
+		glog.V(3).Infof("Calling network plugin %s to tear down pod for %s", r.networkPlugin.Name(), kubecontainer.BuildPodFullName(pod.Name, pod.Namespace))
+		pluginPodID := buildContainerID(&containerID{podID.uuid, pod.Name})
+		if err := r.networkPlugin.TearDownPod(pod.Namespace, pod.Name, pluginPodID); err != nil {
```

Comment:
@yifan-gu the Kubernetes pod UID gets used instead of the rkt UUID.  The Kubernetes UID is already seen extensively in kubelet logs and the API, so I think it's OK to use it instead of the rkt UUID.  With the latest commits, the Kubernetes UID is also written to the service file, so it's possible to correlate the two without looking at logs.